src/mrsimulator/methods/base.py

```diff
@@ -145,57 +145,25 @@ def update(cls, **kwargs):
     def check_method_compatibility(cls, py_dict):
         """Check for events attribute inside the spectral_dimensions. Events are not
         allowed for NamedMethods."""
-        if not isinstance(py_dict["spectral_dimensions"][0], dict):
-            return cls.check_when_arg_is_object(py_dict)
-
         default_method = cls.update(**py_dict)
         default_spectral_dimensions = default_method["spectral_dimensions"]
         for i, item in enumerate(py_dict["spectral_dimensions"]):
-            if item["events"] == [{}]:
+            if isinstance(item, SpectralDimension):
+                item = item.dict()
```

Comment:
Instead of using item.dict(), use item.json(units=False). I think `dict` will return dictionary object with mrsim objects. The `json` function with return nested dictionary with no mrsim objects.

Reply:
`dict` adds all attributes regardless if they are defined whereas `json` only adds attributes which have been defines. Calling `json` on a SpectralDimension without any Events does not include the `events` keyword in the dictionary.

The code can be changed to call `json(units=False)` and check if `events` is present.

Reply:
These lines have been removed